syntax = "proto3";

package product;

service ProductService {
    rpc CheckStock (ProductRequest) returns (ProductResponse);
    rpc GetProduct (ProductIdRequest) returns (ProductDetailResponse);
}

message ProductRequest {
    string id = 1;
    int32 quantity = 2;
}

message ProductIdRequest {
    string id = 1;
}

message ProductResponse {
    bool available = 1;
    double price = 2;
}

message ProductDetailResponse {
    string id = 1;
    string name = 2;
    string slug = 3;
    double price = 4;
}
